### PR TITLE
add optuna dashboard

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hadim** [comment]
@conda-forge/staged-recipes please review

File: recipes/optuna-dashboard/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - bottle
+    - optuna >=2.4
+    - python >=3.6
+    - cython
```
Comment (ocefpaf) [inline recipes/optuna-dashboard/meta.yaml:28]: Does this need cython at run time?

Comment (hadim) [reply recipes/optuna-dashboard/meta.yaml]: Thanks. I have opened a ticket for it https://github.com/conda-forge/optuna-feedstock/issues/23

**Comment by kmaehashi** [comment]
cc: Optuna dev team @g-votte @HideakiImamura @hvy

**Comment by hadim** [comment]
@c-bata: I have added you as a maintainer, in consequence, you'll have full permission on the feedstock repo. There is no need to close this PR. We only need the `cython` dep to be added on the `optuna` package.

(I can also remove myself as a maintainer if that's an issue for you)

**Comment by hadim** [comment]
I have opened a PR on the optuna feedstock: https://github.com/conda-forge/optuna-feedstock/pull/24

**Comment by hadim** [comment]
I guess we just need an optuna release for this recipe to build? (without the cython dep)

**Comment by toshihikoyanase** [comment]
@hadim Thank you for creating a PR for optuna-dashboard.
We just released Optuna v2.7.0, which removed `cython` from the dependencies.
I think the CI errors have been resolved, so could you check it?
https://github.com/optuna/optuna/blob/cfa9011cfc809ad35801e91ff529564bc9b367e1/setup.py#L28-L41
https://github.com/conda-forge/optuna-feedstock/pull/25

**Comment by hadim** [comment]
@conda-forge/staged-recipes please review

File: recipes/optuna-dashboard/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+about:
+  home: https://github.com/optuna/optuna-dashboard
+  summary: Real-time dashboard for Optuna.
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - c-bata
+    - hadim
```
Comment (c-bata) [inline recipes/optuna-dashboard/meta.yaml:48]: > @c-bata: I have added you as a maintainer, in consequence, you'll have full permission on the feedstock repo. There is no need to close this PR. We only need the cython dep to be added on the optuna package.
> 
> (I can also remove myself as a maintainer if that's an issue for you)

@hadim Sorry for the delayed response. Could you update maintainers like the following?

```suggestion
    - c-bata
    - g-votte
    - HideakiImamura
    - hvy
    - iwiwi
    - not522
    - sile
    - smly
    - toshihikoyanase
    - ytsmiling
```

**Comment by hadim** [comment]
@c-bata : all the maintainers listed need to explicitly approve to be a maintainer in this ticket.

File: recipes/optuna-dashboard/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    - c-bata
+    - g-votte
+    - HideakiImamura
+    - hvy
+    - iwiwi
+    - not522
+    - sile
+    - smly
+    - toshihikoyanase
+    - ytsmiling
```
Comment (jakirkham) [inline recipes/optuna-dashboard/meta.yaml:56]: @c-bata @g-votte @HideakiImamura @hvy @iwiwi @not522 @sile @smly @toshihikoyanase @ytsmiling, please let us know here if you are ok being added as maintainers on this feedstock. Thanks! 😄

Comment (c-bata) [reply recipes/optuna-dashboard/meta.yaml:56]: Hi @jakirkham! Thank you for your confirmation. All recipe-maintainers now put a reaction :smile:

**Comment by hadim** [comment]
@conda-forge/staged-recipes please review

File: recipes/optuna-dashboard/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+{% set version = "0.3.1" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/optuna-dashboard-{{ version }}.tar.gz
+  sha256: 277a81b0ebb33f405dfba2cd15d2ec628abd7b4acacd6364c88cea04ee3fa907
```
Comment (c-bata) [inline recipes/optuna-dashboard/meta.yaml:11]: Hi @hadim. I've released the new version of optuna-dashboard. Could you update the version number and sha256 checksum?

```suggestion
{% set version = "0.4.0" %}


package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/optuna-dashboard-{{ version }}.tar.gz
  sha256: 29efd2e1df7e240d64546220a5086980073e2af25b62b2180ad305b5977ab116
```

**Comment by hadim** [comment]
@jakirkham LGTM for me.

File: recipes/optuna-dashboard/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - bottle
+    - optuna >=2.7
+    - python
+    - typing-extensions  # [py<38]
```
Comment (c-bata) [inline recipes/optuna-dashboard/meta.yaml:27]: @hadim Sorry, it is required to add `scikit-learn` here.
https://github.com/optuna/optuna-dashboard/blob/a6af1358bad764a0d60e493ac02a1ee822ed36a0/setup.cfg#L25-L33

```suggestion
    - bottle
    - optuna >=2.7
    - python
    - typing-extensions  # [py<38]
    - scikit-learn
```

**Comment by hadim** [comment]
@conda-forge/staged-recipes please review

**Review by dopplershift — COMMENTED** [review]
Are all of you willing to be maintainers for this recipe?
@g-votte @HideakiImamura @hvy @iwiwi @not522 @sile @smly @toshihikoyanase @ytsmiling

Alternatively, we could merge with only a few and and others could be added in the feedstock.

**Comment by jakirkham** [comment]
Ryan, they all confirmed up here ( https://github.com/conda-forge/staged-recipes/pull/14187#discussion_r607190463 )

**Comment by dopplershift** [comment]
@jakirkham Thanks for letting me know. It was hidden in a "resolved conversation".

**Comment by jakirkham** [comment]
Yep no worries. That's why I shared it 🙂

File: recipes/optuna-dashboard/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
```
[the remark this answers is not in the document]
Comment (c-bata) [reply recipes/optuna-dashboard/meta.yaml:27]: > If that's the case, I would recommend removing the selector and just unconditionally depending on typing-extensions in order to make it noarch: python.

Thank you for your review! It makes sense.

@hadim Could you remove the selector and set `noarch: python`?

[the remark this answers is not in the document]
Comment (hadim) [reply recipes/optuna-dashboard/meta.yaml]: Done.

**Comment by hadim** [comment]
@dopplershift LGTM.

**Review by dopplershift — APPROVED** [review]

**Comment by hadim** [comment]
Thanks all for the help here!

**Comment by jakirkham** [comment]
Thanks for working on this Hadrien! 😄

**Pull request merged**